orchagent/portsorch.cpp

```diff
@@ -2999,7 +3007,7 @@ void PortsOrch::doPortTask(Consumer &consumer)
                         if (!setPortAdvInterfaceTypes(p.m_port_id, adv_interface_types))
                         {
                             SWSS_LOG_ERROR("Failed to set port %s advertised interface type to %s", alias.c_str(), adv_interface_types_str.c_str());
-                            it++;
```

Comment:
I guess the reason to make this change is to avoid retrying. However, I think the problem is we typically treat `false` return value as need retry and this behavior deviate from the expected one. I am not sure what would be the best solution, I think a reasonable one could be making the return value type of `setPortAdvInterfaceTypes` to `task_process_status`. And adding the status of `SAI_STATUS_INVALID_ATTR_VALUE_0` for `SAI_API_PORT` as a special case to return `task_failed` in the failure handling function like https://github.com/Azure/sonic-swss/pull/1815. Then we can make the return value of `task_process_status` type from `handleSaiSetStatus` as the return value of `setPortAdvInterfaceTypes` in case of SAI failure. It follows that we can choose whether to retry here.